### DISCUSSION TITLE
Getting payment required error

### DISCUSSION THREAD

**Discussion by gurkirat1996** [AsamK/signal-cli#510] [Q&A] [answered]
I was trying to implement the signal APIs but the API call if failing with response
`Tuple{v1=402, v2=<html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=utf-8"/>
<title>Error 402 Payment Required</title>
</head>
<body><h2>HTTP ERROR 402 Payment Required</h2>
<table>
<tr><th>URI:</th><td>/v1/accounts/sms/code/+917755057622</td></tr>
<tr><th>STATUS:</th><td>402</td></tr>
<tr><th>MESSAGE:</th><td>Payment Required</td></tr>
<tr><th>SERVLET:</th><td>io.dropwizard.jersey.setup.JerseyServletContainer-57806346</td></tr>
</table>
</body>
</html>
}`

Is there any official documentation for the Signal API? 
I wanted to create something that is similar to your project.

I saw that there was a related bug for the same: https://github.com/AsamK/signal-cli/issues/201
This is the api call that is used to send an OTP to the desired device.
`https://textsecure-service.whispersystems.org/v1/accounts/sms/code/<PHONE_NUMBER>?client=android`

**Comment by AsamK** [comment, marked as answer]
there is no official API documentation. You can look at the libsignal service code in the android repository: https://github.com/signalapp/Signal-Android

Error 402 means CaptchaRequiredException: https://github.com/AsamK/signal-cli/wiki/Registration-with-captcha

**Reply by gurkirat1996** [reply]
Thanks for the quick response @AsamK

**Reply by gurkirat1996** [reply]
Just being curious, how did you know that captcha was required to solve the above issue?

**Reply by AsamK** [reply]
The android signal client maps a 402 response code to a `CaptchaRequiredException`

**Comment by gurkirat1996** [comment]
This worked. Thanks a ton!!!
